Calibration/EcalCalibAlgos/interface/ECALpedestalPCLworker.h

```diff
@@ -34,7 +34,7 @@
 
 class ECALpedestalPCLworker : public DQMEDAnalyzer {
 public:
-  explicit ECALpedestalPCLworker(const edm::ParameterSet &);
+  explicit ECALpedestalPCLworker(const edm::ParameterSet &, const DQMEDAnalyzerGlobalCache*);
```

Comment:
Why does this need `const DQMEDAnalyzerGlobalCache*`? I do not see it being used.

Reply:
If I don't include it, I receive the following error, seemingly related to changing the base class of `ElectronRecalibSuperClusterAssociator` : 

```
>> Building  shared library tmp/slc7_amd64_gcc900/src/Calibration/EcalCalibAlgos/src/CalibrationEcalCalibAlgos/libCalibrationEcalCalibAlgos.so
In file included from /afs/cern.ch/work/a/atishelm/private/ECAL_AlCaDB_Issue/CMSSW_12_2_0_pre2/src/FWCore/Framework/interface/stream/ProducingModuleAdaptor.h:32,
                 from /afs/cern.ch/work/a/atishelm/private/ECAL_AlCaDB_Issue/CMSSW_12_2_0_pre2/src/FWCore/Framework/interface/stream/EDProducerAdaptor.h:25,
                 from /afs/cern.ch/work/a/atishelm/private/ECAL_AlCaDB_Issue/CMSSW_12_2_0_pre2/src/FWCore/Framework/interface/stream/EDProducerBase.h:27,
                 from /afs/cern.ch/work/a/atishelm/private/ECAL_AlCaDB_Issue/CMSSW_12_2_0_pre2/src/FWCore/Framework/interface/stream/implementors.h:31,
                 from /afs/cern.ch/work/a/atishelm/private/ECAL_AlCaDB_Issue/CMSSW_12_2_0_pre2/src/FWCore/Framework/interface/stream/AbilityToImplementor.h:25,
                 from /afs/cern.ch/work/a/atishelm/private/ECAL_AlCaDB_Issue/CMSSW_12_2_0_pre2/src/FWCore/Framework/interface/stream/EDProducer.h:22,
                 from /afs/cern.ch/work/a/atishelm/private/ECAL_AlCaDB_Issue/CMSSW_12_2_0_pre2/src/Calibration/EcalCalibAlgos/interface/ElectronRecalibSuperClusterAssociator.h:10,
                 from /afs/cern.ch/work/a/atishelm/private/ECAL_AlCaDB_Issue/CMSSW_12_2_0_pre2/src/Calibration/EcalCalibAlgos/plugins/SealModule.cc:6:
/afs/cern.ch/work/a/atishelm/private/ECAL_AlCaDB_Issue/CMSSW_12_2_0_pre2/src/FWCore/Framework/interface/stream/makeGlobal.h: In instantiation of 'T* edm::stream::impl::makeStreamModule(const edm::ParameterSet&, const G*) [with T = ECALpedestalPCLworker; G = DQMEDAnalyzerGlobalCache]':
/afs/cern.ch/work/a/atishelm/private/ECAL_AlCaDB_Issue/CMSSW_12_2_0_pre2/src/FWCore/Framework/interface/stream/ProducingModuleAdaptor.h:96:47:   required from 'void edm::stream::ProducingModuleAdaptor<T, M, B>::setupStreamModules() [with T = ECALpedestalPCLworker; M = edm::stream::EDProducerBase; B = edm::stream::EDProducerAdaptorBase]'
/afs/cern.ch/work/a/atishelm/private/ECAL_AlCaDB_Issue/CMSSW_12_2_0_pre2/src/FWCore/Framework/interface/stream/ProducingModuleAdaptor.h:94:12:   required from here
/afs/cern.ch/work/a/atishelm/private/ECAL_AlCaDB_Issue/CMSSW_12_2_0_pre2/src/FWCore/Framework/interface/stream/makeGlobal.h:40:16: error: no matching function for call to 'ECALpedestalPCLworker::ECALpedestalPCLworker(const edm::ParameterSet&, const DQMEDAnalyzerGlobalCache*&)'
   40 |         return new T(iPSet, iGlobal);
      |                ^~~~~~~~~~~~~~~~~~~~~
In file included from /afs/cern.ch/work/a/atishelm/private/ECAL_AlCaDB_Issue/CMSSW_12_2_0_pre2/src/Calibration/EcalCalibAlgos/plugins/SealModule.cc:9:
/afs/cern.ch/work/a/atishelm/private/ECAL_AlCaDB_Issue/CMSSW_12_2_0_pre2/src/Calibration/EcalCalibAlgos/interface/ECALpedestalPCLworker.h:37:12: note: candidate: 'ECALpedestalPCLworker::ECALpedestalPCLworker(const edm::ParameterSet&)'
   37 |   explicit ECALpedestalPCLworker(const edm::ParameterSet &);
      |            ^~~~~~~~~~~~~~~~~~~~~
/afs/cern.ch/work/a/atishelm/private/ECAL_AlCaDB_Issue/CMSSW_12_2_0_pre2/src/Calibration/EcalCalibAlgos/interface/ECALpedestalPCLworker.h:37:12: note:   candidate expects 1 argument, 2 provided
gmake: *** [config/SCRAM/GMake/Makefile.rules:1700: tmp/slc7_amd64_gcc900/src/Calibration/EcalCalibAlgos/plugins/CalibrationEcalCalibAlgosAuto/SealModule.cc.o] Error 1
```

Reply:
That does not make sense since this is the only line that has been changed in the .h and the .cc files. And in the official CMSSW version it compiles just fine. The compile error also states that `ECALpedestalPCLworker::ECALpedestalPCLworker(const edm::ParameterSet&)` is a candidate.